### ISSUE TITLE
ksort unshapes things

### ISSUE THREAD

**Issue by sserbin** [vimeo/psalm#969]
```php
<?php

$a = ['a' => 3, 'b' => 4];
ksort($a);
acceptsAShape($a);

/**
 * @param array{a:int,b:int} $a
 */
function acceptsAShape(array $a): void {}

```

https://getpsalm.org/r/c8327d0fa0

Expected: no errors
Actual: `ERROR: TypeCoercion - 5:15 - Argument 1 of acceptsAShape expects array{a:int, b:int}, parent type array<string(a)|string(b), int(4)|int(3)> provided`

**Comment by muglug** [comment]
Thanks! Fixed in https://github.com/vimeo/psalm/commit/e1bc9c65fc188e2a4240b5c6b6cc5f7a56fc0295
